--- libebml/make/linux/Makefile.orig	2003-07-17 11:47:07.000000000 +0200
+++ libebml/make/linux/Makefile	2003-09-02 15:18:20.727607600 +0200
@@ -25,6 +25,7 @@
 INSTALL_OPTS = -m 644
 INSTALL_OPTS_LIB = -m 644
 INSTALL_DIR_OPTS = -m 755
+LIBTOOL = libtool
 
 # Options
 EXTENSION=.cpp
@@ -45,7 +46,7 @@
 LIBS=
 
 # Names
-LIBRARY=libebml.a
+LIBRARY=libebml.la
 
 # source-files
 sources:=$(wildcard ${SRC_DIR}*$(EXTENSION))
@@ -56,8 +57,8 @@
 # files holding dependency information; replace .cxx extension with .dep
 dependencies:=$(patsubst %$(EXTENSION),%.dep,$(sources))
 
-# object files; replace .cxx extension with .o
-objects:=$(patsubst %$(EXTENSION),%.o,$(sources))
+# object files; replace .cxx extension with .lo
+objects:=$(patsubst %$(EXTENSION),%.lo,$(sources))
 
 DEPENDFLAGS  = ${CXXFLAGS} ${INCLUDE}
 
@@ -66,14 +67,12 @@
 lib library: $(LIBRARY)
 
 # Build rules
-%.o: %$(EXTENSION)
-	$(CXX) -c $(CXXFLAGS) $(INCLUDE) -o $@ $<
+%.lo: %$(EXTENSION)
+	$(LIBTOOL) --mode=compile $(CXX) -c $(CXXFLAGS) $(INCLUDE) -o $@ $<
 
 $(LIBRARY): $(objects)
-	$(AR) $@ $(objects)
-	$(RANLIB) $@
-#	$(LD) $(LDFLAGS) -o $@ $^ $(LIBS)
-#	${LD} -o $@ ${LDFLAGS} ${OBJS} ${LIBS} ${EXTRA_LIBS}
+	$(LIBTOOL) --mode=link $(LD) $(LDFLAGS) -o $@ $^ $(LIBS) -rpath $(libdir)
+
 
 clean:
 	rm -f $(objects)
@@ -86,15 +85,15 @@
 	$(DEPEND) $(DEPENDFLAGS) $(sources)
 
 install: $(LIBRARY)
-	$(INSTALL) $(INSTALL_DIR_OPTS) -d $(libdir)
-	$(INSTALL) $(INSTALL_OPTS_LIB) $(LIBRARY) $(libdir)
-	$(INSTALL) $(INSTALL_DIR_OPTS) -d $(includedir)
+	$(INSTALL) $(INSTALL_DIR_OPTS) -d $(DESTDIR)$(libdir)
+	$(LIBTOOL) --mode=install $(INSTALL) $(LIBRARY) $(DESTDIR)$(libdir)
+	$(INSTALL) $(INSTALL_DIR_OPTS) -d $(DESTDIR)$(includedir)
 	for i in $(INCLUDE_DIR)/*.h; do \
-		$(INSTALL) $(INSTALL_OPTS) $$i $(includedir) ; \
+		$(INSTALL) $(INSTALL_OPTS) $$i $(DESTDIR)$(includedir) ; \
 	done
-	$(INSTALL) $(INSTALL_DIR_OPTS) -d $(includedir)/c
+	$(INSTALL) $(INSTALL_DIR_OPTS) -d $(DESTDIR)$(includedir)/c
 	for i in $(INCLUDE_DIR)/c/*.h; do \
-		$(INSTALL) $(INSTALL_OPTS) $$i $(includedir)/c ; \
+		$(INSTALL) $(INSTALL_OPTS) $$i $(DESTDIR)$(includedir)/c ; \
 	done
 
 # DO NOT DELETE
