--- libebml-0.7.8/make/linux/Makefile~	2010-08-05 22:53:47.000000000 +0300
+++ libebml-0.7.8/make/linux/Makefile	2010-08-05 22:54:21.405342023 +0300
@@ -82,10 +82,10 @@
 
 # Build rules
 %.lo: %$(EXTENSION)
-	$(LIBTOOL) --mode=compile $(CXX) -c $(COMPILEFLAGS) -o $@ $<
+	$(LIBTOOL) --mode=compile --tag=CC $(CXX) -c $(COMPILEFLAGS) -o $@ $<
 
 $(LIBRARY): $(objects)
-	$(LIBTOOL) --mode=link $(LD) $(LDFLAGS) -o $@ $^ $(LIBS) -rpath $(libdir)
+	$(LIBTOOL) --mode=link --tag=CC $(LD) $(LDFLAGS) -o $@ $^ $(LIBS) -rpath $(libdir)
 
 clean:
 	rm -f $(objects) $(objects_so)
